### DISCUSSION TITLE
create Node dynamically in Angular Treeview  component

### DISCUSSION THREAD

**Discussion by pmullot** [vmware-clarity/ng-clarity#207] [General]
I have a component with a Treeview which is supposed to represent a complex js object.
For each node of the tree, there should be several "normal" nodes (representing the underlying object properties) and other nodes under which new nodes could appear. Those new nodes will have the same form as their parent. It's a recursive tree.

The issue is that those new Nodes, under any of the "Children" nodes, are added dynamically by the user when clicking a button.

The idea is that clicking on the + blue button, a new node will be created under "sub-node 1", which in turn will also have the 2 properties, plus the possibility to add its own subnode
![image](https://user-images.githubusercontent.com/75703/176728825-a929ee98-ca41-45e7-a5d3-fd8d66c46513.png)

I've built this example using the `*clrRecursiveFor` directive. If I simply add a new node as 
```
{
        "type": "children",
        "label": "sub-sub-node 1",
        "children": [
          {
            "type": "prop",
            "label": "Color: Red"
          },
          {
            "type": "prop",
            "label": "Size: Big"
          }
        ]
      }
```
to the "children" node of the one called "sub-node 1", nothing happens on the tree. The underlying object is correctly updated but the tree is not "refreshed"

Can someone give me a hint on how to implement this?

Thanks!

**Comment by kevinbuhmann** [comment]
I made a proof of concept of this. Check it out at https://stackblitz.com/edit/clarity-light-theme-clr13-cds6-bl6ytw?file=src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html.

**Reply by pmullot** [reply]
That's awesome!
Thanks a lot for the solution and the speed!
Really appreciated!

And congrats for the library, It's fantastic!
